subsys/net/lib/wifi_mgmt_ext/wifi_mgmt_ext.c

```diff
@@ -0,0 +1,181 @@
+
+LOG_MODULE_REGISTER(wifi_mgmt_ext, CONFIG_WIFI_MGMT_EXT_LOG_LEVEL);
+
+static inline const char *wpa_supp_security_txt(enum wifi_security_type security)
+{
+	switch (security) {
+	case WIFI_SECURITY_TYPE_NONE:
+		return "OPEN";
+	case WIFI_SECURITY_TYPE_PSK:
+		return "WPA-PSK";
+	case WIFI_SECURITY_TYPE_PSK_SHA256:
+		return "WPA-PSK-SHA256";
+	case WIFI_SECURITY_TYPE_SAE:
+		return "SAE";
+	case WIFI_SECURITY_TYPE_UNKNOWN:
+	default:
+		return "UNKNOWN";
+	}
+}
+
+static int add_network_from_credentials_struct_personal(struct wifi_credentials_personal *creds)
+{
+	int ret = 0;
+	struct add_network_resp resp = {0};
+
+	ret = wpa_cli_api_add_network(&resp);
+	if (ret) {
+		LOG_ERR("Encoding error: %d returned at %s:%d", ret, __FILE__, __LINE__);
+		return ret;
+	}
+
+	call_wpa_cli("set_network %d ssid \"%s\"", resp.network_id, creds->header.ssid);
+	call_wpa_cli("set_network %d key_mgmt %s", resp.network_id,
+		     wpa_supp_security_txt(creds->header.type));
+
+	if (creds->header.type == WIFI_SECURITY_TYPE_PSK ||
+	    creds->header.type == WIFI_SECURITY_TYPE_PSK_SHA256) {
+		call_wpa_cli("set_network %d psk \"%s\"", resp.network_id, creds->password);
+	}
+
+	if (creds->header.type == WIFI_SECURITY_TYPE_SAE) {
+		call_wpa_cli("set_network %d sae_password \"%s\"",
+			     resp.network_id, creds->password);
+	}
+
+	if (creds->header.flags & WIFI_CREDENTIALS_FLAG_BSSID) {
+		call_wpa_cli("set_network %d bssid %02hhx:%02hhx:%02hhx:%02hhx:%02hhx:%02hhx",
+			resp.network_id,
+			creds->header.bssid[0], creds->header.bssid[1],
+			creds->header.bssid[2], creds->header.bssid[3],
+			creds->header.bssid[4], creds->header.bssid[5]);
+	}
+
+	if (creds->header.flags & WIFI_CREDENTIALS_FLAG_FAVORITE) {
+		call_wpa_cli("set_network %d priority 1", resp.network_id);
+	}
+
+	if (creds->header.flags & WIFI_CREDENTIALS_FLAG_2_4GHz) {
+		call_wpa_cli("set_network %d freq_list "
+			     "2412 2417 2422 2427 2432 2437 2442 "
+			     "2447 2452 2457 2462 2467 2472 2484", resp.network_id);
+	}
+
+	if (creds->header.flags & WIFI_CREDENTIALS_FLAG_5GHz) {
+		call_wpa_cli("set_network %d freq_list "
+			     "5180 5200 5220 5240 5260 5280 5300 "
+			     "5310 5320 5500 5520 5540 5560 5580 "
+			     "5600 5620 5640 5660 5680 5700 5745 "
+			     "5765 5785 5805 5825", resp.network_id);
+	}
+
+	call_wpa_cli("enable_network %d", resp.network_id);
+	return ret;
+}
+
+static void add_stored_network(void *cb_arg, const char *ssid, size_t ssid_len)
+{
+	int ret = 0;
+	struct wifi_credentials_personal creds;
+
+	/* load stored data */
+	ret = wifi_credentials_get_by_ssid_personal_struct(ssid, ssid_len, &creds);
+
+	if (ret) {
+		LOG_ERR("Loading WiFi credentials failed for SSID [%.*s], len: %d, err: %d",
+			ssid_len, ssid,  ssid_len, ret);
+		return;
+	}
+
+	add_network_from_credentials_struct_personal(&creds);
+}
+
+static int add_static_network_config(void)
+{
+	if (strnlen(CONFIG_WIFI_CREDENTIALS_STATIC_SSID, WIFI_SSID_MAX_LEN) > 0) {
```

Comment:
nit, Kconfig strings will always be NULL-terminated, so simple `strlen` should do.

Reply:
Thanks for the hint! I'd like to keep the `strnlen` though. Just for peace-of-mind reasons.